### ISSUE TITLE
aws-sdk to use exact dependecies for other aws-sdk package dependencies

### ISSUE THREAD

**Issue by Amplifiyer** [aws/aws-sdk-js-v3#1072]
**Is your feature request related to a problem? Please describe.**

Now since aws-sdk has many packages that depend on each other and use loose version numbers (^ versions), clients using a particular version of aws-sdk-v3 clients may end up getting newer versions of transitive dependencies of aws-sdk-v3 if they don't use lock files or delete lock files.

**Describe the solution you'd like**
I'd like aws-sdk clients to use exact versions for all their other aws-sdk package dependencies (such that the closure of aws-sdk packages for a given version/release is always compatible)

This will ideally decouple aws-amplify (or other clients) to be always aware of any aws-sdk-v3 releases

**Comment by trivikr** [comment]
CDK is also facing similar issue, and they've discussed this in https://github.com/aws/aws-cdk-rfcs/issues/6

**Comment by Amplifiyer** [comment]
@trivikr thanks for linking the issue. Can you summarize that issue and how JS-SDK can leverage that solution?

**Comment by trivikr** [comment]
> Can you summarize that issue and how JS-SDK can leverage that solution?

As posted in https://github.com/aws/aws-cdk-rfcs/pull/122#issuecomment-610961073, the issue is a concern for CDK as it's a "framework" and mostly consumed by "CDK Apps". SDKs are mostly used as libraries where there's less dependency in-between clients.

We're evaluating the CDK RFC, and will provide more details.

**Comment by trivikr** [comment]
Refs: https://github.com/aws/aws-cdk-rfcs/pull/122#issuecomment-592153193

> The main reason the SDKs are bundled separately are to reduce the runtime footprint and that’s a legitimate concern when it comes to a library you link against.

**Comment by trivikr** [comment]
Closing as we use exact dependencies from beta.5 onwards
https://github.com/aws/aws-sdk-js-v3/releases/tag/v1.0.0-beta.5
